Source/WebCore/rendering/RenderWidget.cpp

```diff
@@ -257,12 +257,15 @@ void RenderWidget::paintContents(PaintInfo& paintInfo, const LayoutPoint& paintO
     // to paint itself. That way it will composite properly with z-indexed layers.
     LayoutRect paintRect = paintInfo.rect;
 
-    OptionSet<PaintBehavior> oldBehavior = PaintBehavior::Normal;
-    if (paintInfo.paintBehavior & PaintBehavior::DefaultAsynchronousImageDecode) {
-        if (RefPtr frameView = dynamicDowncast<LocalFrameView>(m_widget)) {
-            oldBehavior = frameView->paintBehavior();
-            frameView->setPaintBehavior(oldBehavior | PaintBehavior::DefaultAsynchronousImageDecode);
-        }
+    static constexpr OptionSet flagsToCopy {
+        PaintBehavior::DefaultAsynchronousImageDecode,
+        PaintBehavior::FlattenCompositingLayers,
+        PaintBehavior::Snapshotting
+    };
+    std::optional<OptionSet<PaintBehavior>> oldBehavior = PaintBehavior::Normal;
+    if (RefPtr frameView = dynamicDowncast<LocalFrameView>(m_widget)) {
+        oldBehavior = frameView->paintBehavior();
+        frameView->setPaintBehavior(*oldBehavior | (paintInfo.paintBehavior & flagsToCopy));
```

Comment:
maybe name it `oldFrameBehavior` for slightly more clarity, since there's multiple behaviors involved here